bluemira/power_cycle/base.py

```diff
@@ -0,0 +1,136 @@
+"""
+Base classes for the power cycle model.
+"""
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from bluemira.base.error import BluemiraError
+
+
+class PowerCycleError(BluemiraError):
+    """
+    Exception class for Power Cycle classes.
+
+    Class should be named in the following format: '(Class)Error',
+    where '(Class)' is the name of the associated class (source) for
+    which the exception class is being defined.
+    """
+
+    def __init__(self, case=None, msg=""):
+        message = msg
+        if case:
+            all_errors = self._errors()
+            if case in all_errors:
+                extra_msg = str(all_errors[case])
+                message = message + " " + extra_msg
+            else:
+                raise ValueError(
+                    f"The requested error case '{case}' has not been "
+                    f"defined for the error class '{self.__class__}'."
+                )
+        super().__init__(message)
+
+    def _source(self):
+        class_name = self.__class__
+        source_class = class_name.replace("Error", "")
+        source_class = "'" + source_class + "'"
+        return source_class
+
+    @abstractmethod
+    def _errors(self) -> Dict:
+        pass
+
+
+class PowerCycleABCError(PowerCycleError):
+    """
+    Exception class for 'PowerCycleABC' class of the Power Cycle module.
+    """
+
+    def _errors(self):
+        errors = {
+            "name": [
+                "Invalid 'name' parameter. The 'name' attribute of ' "
+                f"instances of the {self._source} class must be of "
+                "the 'str' class.",
+            ],
```

Comment:
strings auto join in a list and a tuple if you remove the last command put it in a tuple it will just be a string
```suggestion
            "name": (
                "Invalid 'name' parameter. The 'name' attribute of ' "
                f"instances of the {self._source} class must be of "
                "the 'str' class."),
```